modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp

```diff
@@ -167,7 +224,114 @@ TEST(StatefulKernel, InvalidReallocatingKernel)
     cv::GComputation comp(cv::GIn(in), cv::GOut(out));
 
     EXPECT_THROW(comp.apply(in_mat, out_mat, cv::compile_args(pkg)), std::logic_error);
+}
+
+namespace
+{
+    void compareBackSubResults(const cv::Mat &actual, const cv::Mat &expected,
+                                       const int diffPercent)
+    {
+        GAPI_Assert(actual.size() == expected.size());
+        int allowedNumDiffPixels = actual.size().area() * diffPercent / 100;
+
+        cv::Mat diff;
+        cv::absdiff(actual, expected, diff);
+        #undef countNonZero
```

Comment:
This `#undef` has an effect for the rest of the file, unfortunately.

@alalek if we really need countNonZero, what should we do?

@AsyaPronina instead you can calculate a histogram over absdiff.

Reply:
Calculated via histogram